android/src/main/java/com/learnium/RNDeviceInfo/resolver/DeviceIdResolver.java

```diff
@@ -1,46 +1,80 @@
 package com.learnium.RNDeviceInfo.resolver;
 
 import android.content.Context;
+import android.content.SharedPreferences;
+import android.os.Build;
 
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
+import java.util.UUID;
+
+import com.facebook.react.bridge.ReactApplicationContext;
+
+import com.learnium.RNDeviceInfo.RNDeviceModule;
 
 /**
  * Instance Id resolver's single purpose is to get the device's Instance Id
  * author: Andres Aguilar
  */
 public class DeviceIdResolver {
 
-  private final Context context;
+  private final ReactApplicationContext context;
```

Comment:
Why was the type changed from `Context` to `ReactApplicationContext` ? I may just be out of the loop

Reply:
What I really needed is the SharedPreferences object, which, if *I* understand correctly is normally fetched via

```java
SharedPreferences sharedPref = getActivity().getPreferences(Context.MODE_PRIVATE);
```

What I don't have normally is the `Activity`

- you can access it from the module via an instance method but we are in a separate object here
- you have to be careful because frequently the Activity is null when you least expect it! Classic react-native trap - the bridge may dispose of the Activity as app goes to background, but code is executing async and you get a null

What I want is the shared prefs though, and it turns out the `ReactApplicationContext` has that as a direct method, and that is what we get during module construction.

So I plumb that through to this class during instantiation, and now I can get the shared prefs.

Okay so then I did abstract the actual shared prefs get to a static on the main module, but that's so they were all in one spot without duplicating the constant and access style everywhere. But it still requires that ReactApplicationContext to get them

It's a little janky, but it works and does not depend on the Activity (which I don't always have)

Reply:
It looks like this is actually available from *any* Context, so you're right, I don't really need to change the type here, I can keep it more generic

https://developer.android.com/reference/android/content/Context#getSharedPreferences(java.lang.String,%20int)

Reply:
@schie pushed 3rd commit with use of `Context` vs `ReactApplicationContext`